src/main/thrift/parquet.thrift

```diff
@@ -467,6 +467,35 @@ struct SchemaElement {
   10: optional LogicalType logicalType
 }
 
+struct SchemaElementV3 {
+  /** Data type for this field. */
+  1: optional Type type;
+
+  /** If type is FIXED_LEN_BYTE_ARRAY, this is the byte length of the values.
+   *
+   * CHANGED from v1: this must be omitted for other column types.
+   */
+  2: optional i32 type_length;
+
+  /** repetition of the field. */
+  3: optional FieldRepetitionType repetition_type;
+
+  /** Name of the field in the schema */
+  4: required string name;
+
+  /** Nested fields. */
+  5: optional i32 num_children;
+
+  /** CHANGED from v1: from i32 to i64
```

Comment:
have you encountered a use-case for this?  2 Billion field ID seems quite high?

Reply:
Same question!

Reply:
Ok. It seemed a bit arbitrary to limit the width of this, but I can undo the change.

Reply:
I think it is backwards compatible and in the long run, it probably isn't a big deal either way.  I as just curious if there as a reason?